board/linux/README.md

```diff
@@ -0,0 +1,110 @@
+
+# board-linux
+
+This directory contains the necessary files for the linux simulator filesystem. To have this same filesystem interface for the linux simulator
+and rt1051 we have to translate paths. It has been done by wrappers which translate path and then invoke real syscall from
+`glibc`. The advantage of this solution is that we don't need to translate the path in the application layer, everything is done by low-level
+wrappers. So we do not need to modify third-party libraries to translate path. However, we need to maintain `glibc`'s versions if we change the compiler or libraries. 
+
+
+## How does it work
+
+If `opendir` function is invoked 
+``` c
+DIR *dir;
+if ((dir = opendir (".")) == NULL) {
+    perror ("Cannot open .");
+    exit (1);
+}
+//...
+```
+it is linked to our wrapper as `_iosys_opendir`
+``` c
+DIR *_iosys_opendir(const char *dirname)
+{
+    __dirstream *ret{};
+    if (!dirname) {
+        //...
+        errno = EINVAL;
+        return ret;
+    }
+    if (vfs::redirect_to_image(dirname)) {
+        //...
+    }
+    else {
+        char tmp[PATH_MAX];
+        // Path transtation
+        const auto newpath = vfs::npath_translate(dirname, tmp);
+        // Invoke opendir from glibc
+        ret = real::opendir(newpath);
+    }
+    return ret;
+}
+__asm__(".symver _iosys_opendir,opendir@GLIBC_2.2.5");
+```
+As you see it is done by `.symver _iosys_opendir,opendir@GLIBC_2.2.5` which links it with a proper version of `glibc` function.
+
+```
+        +--------------+                      +--------------------------+                  +-----------------+
+        |              |    invoke wrapper    | _iosys_opendir(...) {    |   invoke glibc   |                 |
+        | opendir(...) |--------------------->| opendir(translated_path) |----------------->| real::opendir() |
+        |              |                      | }                        |                  |                 |
+        +--------------+                      +--------------------------+                  +-----------------+
+```
+
+## How to find correct version
+Our library does not handle all syscalls because we don't know the exact syscall version. While changing compiler or some library
```

Comment:
The style here could be more consistent - e.g. "don't" vs. "do not", prepositions - but I won't insist :)

Reply:
Done